In [41]:
import pandas as pd

df = pd.read_csv(
    "../data/DSI_kickstarterscrape_dataset.csv",
    encoding="latin1"
)

print("Dataset shape:", df.shape)

Dataset shape: (45957, 17)


In [42]:
print("Columns:")
print(df.columns.tolist())

print("\nFirst 20 location values:")
print(df["location"].head(20))

print("\nMissing location values:")
print(df["location"].isna().sum())

Columns:
['project id', 'name', 'url', 'category', 'subcategory', 'location', 'status', 'goal', 'pledged', 'funded percentage', 'backers', 'funded date', 'levels', 'reward levels', 'updates', 'comments', 'duration']

First 20 location values:
0        Columbia, MO
1       Maplewood, NJ
2     Los Angeles, CA
3     Los Angeles, CA
4            Novi, MI
5        Portland, OR
6     Collegedale, TN
7         Chicago, IL
8         Chicago, IL
9         Chicago, IL
10      Nashville, TN
11                NaN
12        Ashland, OR
13       New York, NY
14                NaN
15        Chicago, IL
16     Washington, DC
17         Gretna, NE
18          Boise, ID
19          Tampa, FL
Name: location, dtype: str

Missing location values:
1322


In [43]:
print(df["location"].dropna().sample(20, random_state=42).to_string(index=False))

       Milwaukee, WI
        Syracuse, NY
  Salt Lake City, UT
     Little Rock, AR
         Atlanta, GA
   San Francisco, CA
       Nashville, TN
         Atlanta, GA
Melbourne, Australia
         Chicago, IL
    East Village, NY
         Seattle, WA
      Greenville, SC
        New York, NY
         Atlanta, GA
          Canton, OH
     Los Angeles, CA
         Brevard, NC
          Athens, OH
   Mae Sot, Thailand


In [44]:
us_states = {
    "AL", "AK", "AZ", "AR", "CA", "CO", "CT", "DE",
    "FL", "GA", "HI", "ID", "IL", "IN", "IA", "KS",
    "KY", "LA", "ME", "MD", "MA", "MI", "MN", "MS",
    "MO", "MT", "NE", "NV", "NH", "NJ", "NM", "NY",
    "NC", "ND", "OH", "OK", "OR", "PA", "RI", "SC",
    "SD", "TN", "TX", "UT", "VT", "VA", "WA", "WV",
    "WI", "WY", "DC"
}

location_state = (
    df["location"]
    .fillna("")
    .str.rsplit(",", n=1)
    .str[-1]
    .str.strip()
)

us_df = df[location_state.isin(us_states)].copy()

print("Original records:", len(df))
print("U.S. records:", len(us_df))
print("Records not identified as U.S.:", len(df) - len(us_df))

Original records: 45957
U.S. records: 42268
Records not identified as U.S.: 3689


In [45]:
print("U.S. dataset shape:", us_df.shape)

print("\nUnique U.S. location values:")
print(us_df["location"].nunique())

print("\nSample of U.S. locations:")
print(us_df["location"].sample(20, random_state=42).to_string(index=False))

U.S. dataset shape: (42268, 17)

Unique U.S. location values:
4035

Sample of U.S. locations:
  Los Angeles, CA
       Boston, MA
       Boston, MA
      Augusta, GA
   Palm Beach, FL
    San Diego, CA
San Francisco, CA
       Austin, TX
   Greenville, SC
 Philadelphia, PA
    Charlotte, NC
      Seattle, WA
    Hollywood, CA
      Chicago, IL
       Lowell, MA
       Boston, MA
   Chesapeake, VA
        Scott, LA
        Miami, FL
     New York, NY


In [46]:
# Verify that all filtered locations end with a valid U.S. state/DC abbreviation

us_location_states = (
    us_df["location"]
    .str.rsplit(",", n=1)
    .str[-1]
    .str.strip()
)

invalid_states = sorted(set(us_location_states) - us_states)

print("Invalid state values:", invalid_states)
print("Number of invalid state values:", len(invalid_states))

Invalid state values: []
Number of invalid state values: 0


In [47]:
print("Sample of U.S.-based campaigns:")
print(us_df[["name", "location", "status"]].head(20).to_string(index=False))

Sample of U.S.-based campaigns:
                                                            name        location     status
                                           WHILE THE TREES SLEEP    Columbia, MO successful
                            Educational Online Trading Card Game   Maplewood, NJ     failed
                                                           STRUM Los Angeles, CA       live
     GETTING OVER - One son's search to finally know his father. Los Angeles, CA successful
The Launch of FlyeGrlRoyalty &quot;The New Name In Fashion&quot;        Novi, MI     failed
     Dinner Party - a short film about friendship... and revenge    Portland, OR successful
                                                           Mezzo Collegedale, TN     failed
         Help APORTA continue to make handwoven/knit accessories     Chicago, IL successful
                                         Music - Comedy - Album!     Chicago, IL successful
                                         The Apo

## Task 4: Prepare U.S.-Based Campaign Records

### Location Review

The dataset contains 45,957 campaign records and 17 columns. The `location` column was reviewed to understand how campaign locations are represented. Location values generally follow a `City, State/Country` format, such as `Los Angeles, CA` and `Melbourne, Australia`.

There are 1,322 records with missing location values. These records were not included in the U.S.-based subset because a U.S. location could not be determined from the available location information.

### U.S.-Based Campaign Filtering

To identify U.S.-based campaigns, the final component of each location value was extracted and compared against a list containing the 50 U.S. state abbreviations and Washington, D.C.

The filtered dataset contains:

- **Original dataset:** 45,957 records
- **U.S.-based campaigns:** 42,268 records
- **Records not identified as U.S.-based:** 3,689 records
- **Unique U.S. location values:** 4,035

The resulting `us_df` DataFrame contains only campaign records whose location ends with a recognized U.S. state or Washington, D.C. abbreviation.

### Location Information Review

A sample of the filtered records was reviewed using the campaign name, location, and status fields. The sample showed U.S.-based locations including Columbia, MO; Los Angeles, CA; Chicago, IL; New York, NY; and Austin, TX.

The original `location` field was retained rather than creating duplicate city and state columns. This preserves the original location information for potential use in future exploratory analysis.

### Verification

The U.S.-based dataset was verified by checking the state/DC portion of each retained location against the list of recognized U.S. state and D.C. abbreviations.

**Verification result:**

- **Invalid state values:** 0
- **U.S. dataset shape:** 42,268 rows × 17 columns

The resulting U.S.-based campaign dataset is prepared for future exploratory analysis and machine learning.